src/core_plugins/choropleth/public/choropleth_controller.js

```diff
@@ -0,0 +1,133 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import AggConfigResult from 'ui/vis/agg_config_result';
+import KibanaMap from 'ui/vis_maps/kibana_map';
+import FilterBarFilterBarClickHandlerProvider from 'ui/filter_bar/filter_bar_click_handler';
+import ChoroplethLayer from './choropleth_layer';
+import colorramps from 'ui/vislib/components/color/colormaps';
+import AggResponsePointSeriesTooltipFormatterProvider from './tooltip_formatter';
+
+const module = uiModules.get('kibana/choropleth', ['kibana']);
+module.controller('KbnChoroplethController', function ($scope, $element, Private, getAppState, tilemapSettings) {
+
+  const filterBarClickHandler = Private(FilterBarFilterBarClickHandlerProvider);
+  const containerNode = $element[0];
+  const tooltipFormatter = Private(AggResponsePointSeriesTooltipFormatterProvider);
+
+  let kibanaMap = null;
+  let choroplethLayer = null;
+
+  async function makeKibanaMap() {
+
+    if (!tilemapSettings.isInitialized()) {
+      await tilemapSettings.loadSettings();
+    }
+
+    const minMaxZoom = tilemapSettings.getMinMaxZoom(false);
+    kibanaMap = new KibanaMap(containerNode, minMaxZoom);
+    const url = tilemapSettings.getUrl();
+    const options = tilemapSettings.getTMSOptions();
+    kibanaMap.setBaseLayer({
+      baseLayerType: 'tms',
+      options: { url, ...options }
+    });
+    kibanaMap.addLegendControl();
+  }
+
+  const kibanaMapReady = makeKibanaMap();
+  $scope.$watch('esResponse', async function (response) {
```

Comment:
Since we aren't using `await` below, we technically don't need the `async` keyword. However, I'd argue for using `await` instead of removing the `async`